Various improvements to flockeropenssl scripts

sgnn7 opened the pull request:
https://phabricator.endlessm.com/T11072

sgnn7 committed:
Improved the argument parsing and help text

Old text wasn't very intuitive so it got a bit improved as well as some
conditions cleaned up to be more readable.

sgnn7 committed:
Cleaned up the script a bit

Code is now easier to read and maintain, we have key-length variable and
a bit cleaner way to create the certs.

sgnn7 committed:
Extracted common functions

We kept reusing the same constructs which wasn't needed so now it's all
compact.

sgnn7 committed:
Simplified function usage even more

Simpler & more streamlined but it could still use a bit more cleanup.

sgnn7 committed:
Better organization of files added

We now store all cluster-relevant files in the folder for that
particular cluster name.

sgnn7 committed:
Improved output

Output is a bit easier to visually parse

sgnn7 committed:
Added checking of existence of old certificates

If we alreday have generated certificates, we want to make the user
aware of it and have them force the overwrite.

sgnn7 committed:
Updated README

It was way out of date

sgnn7 committed:
Added a gitignore file

We had a lot of useless files being tracked for changes

sgnn7 committed:
Added ability to invoke the script from outside of checkout directoy

Old code wasn't sturdy enough to handle those cases

sgnn7 committed:
Allow output of files to go to an output dir of our choosing

This makes the whole thing quite a bit more convenient than before.

sgnn7 committed:
Fixed healp output formatting

Old code wasn't outputting properly the help string when invoked with
`-h`

sgnn7 commented:
@wallnerryan Tested on an internal cluster with all the changes and things seem to be running smoothly.

wallnerryan commented on README.md:
````diff
@@ -2,59 +2,34 @@
 You can view help message by
 ```
 $ ./flockeropenssl/generate_flocker_certs.sh -h
+Usage: $0 (-i=<control_ip> | -d=<control_fqdn>) [-f=openssl_conf] -c=<cluster_name> -n=<node>[,<node> ... ]
````
thanks this is cleaner

wallnerryan commented on generate_flocker_certs.sh:
```diff
@@ -39,133 +39,256 @@ HELP_MSG="""
-# generate the CA:
-# ----------------------------------------
+echo "Cleaning up old CA dirs"
+rm -rf ssl
```
removing things makes me nervous, especially if someone uses "ssl" outside of this script. Which could be the case if its not `./ssl` Maybe rename this to `./flockerssl` so its obvious.  Same in the openssl conf file.

wallnerryan commented on README.md:
````diff
@@ -63,51 +38,34 @@ $ ./flockeropenssl/generate_flocker_certs.sh -h
 
 ```
-./flockeropenssl/generate_flocker_certs.sh -d="ec2-52-91-11-106.compute-1.amazonaws.com" -n="ec2-52-91-11-106.compute-1.amazonaws.com,node2,node3" -f=/etc/flocker/ssl/flockeropenssl/openssl.cnf
+./flockeropenssl/generate_flocker_certs.sh -d=www.foobar.com -k=1024 -c=staging-us-west-2 -n=one,two
````
make this more generic, instead of amazon-like azs. "staging-1" for instance.

wallnerryan commented on generate_flocker_certs.sh:
```diff
@@ -39,133 +39,256 @@ HELP_MSG="""
+    echo "Forcefuly removing old data from old cluster"
+    rm -rf "$output_dir"
+  else
+    echo "ERROR! Cluster already created and '--force' not specified! Exiting!"
```
thanks, this is nice.

wallnerryan commented:
@sgnn7  Looks great, thanks, few comments. Also we're moving this to https://github.com/ClusterHQ/flocker-openssl.  can you make the above changes and re-submit to the new locations, then close this PR.

wallnerryan commented on generate_flocker_certs.sh:
```diff
@@ -1,36 +1,36 @@
 # and combined with the implementation in flocker/flocker/ca/_ca.py
 #
-# authors: Brendan Cox "justnoise" , and Ryan Wallner "wallnerryan"
+# Authors:
```
mind splitting this out into CONTRIBUTORS.md?

sgnn7 committed:
Changed prefix to flockerssl

This is to make sure we don't overwrite anyone's real CA SSL directories
or accidentally remove them.

sgnn7 committed:
Genericized the example server name

Not AWS-specific anymore

sgnn7 committed:
Moved authors to AUTHORS.md

Requested split into this separate file

sgnn7 commented:
@wallnerryan Closing PR to address changes and to reopen it for new repo (https://github.com/ClusterHQ/flocker-openssl)
